src/extension.ts

```diff
@@ -149,6 +149,8 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
                 node = await ext.tree.showTreeItemPicker<MongoDocumentTreeItem | DocDBDocumentTreeItem>([MongoDocumentTreeItem.contextValue, DocDBDocumentTreeItem.contextValue], actionContext);
             }
 
+            // Clear un-uploaded local changes to the document before opening https://github.com/microsoft/vscode-cosmosdb/issues/1619
+            await node.refresh();
```

Comment:
Our extension isn't caching the local changes, so you don't need to refresh the whole node. Pretty sure VS Code is doing the caching, in which case this should work:
```
ext.fileSystem.fireChangedEvent(node);
```

We call `fireChangedEvent` in `refreshImpl`, which I'm pretty sure is the only reason `node.refresh()` worked in the first place.